AnnService/inc/Core/Common/Labelset.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _SPTAG_COMMON_LABELSET_H_
+#define _SPTAG_COMMON_LABELSET_H_
+
+#include <atomic>
+#include "Dataset.h"
+
+namespace SPTAG
+{
+    namespace COMMON
+    {
+        class Labelset
+        {
+        private:
+            std::atomic<SizeType> m_deleted;
+            Dataset<std::int8_t> m_data;
+            
+        public:
+            Labelset() 
+            {
+                m_deleted = 0;
+                m_data.SetName("DeleteID");
+            }
+
+            void Initialize(SizeType capacity)
+            {
+                m_data.Initialize(capacity, 1);
+            }
+
+            inline size_t Count() const { return m_deleted.load(); }
+
+            inline bool Contains(const SizeType& key) const
+            {
+                return *m_data[key] == 1;
+            }
+
+            inline void Insert(const SizeType& key)
+            {
+                if (*m_data[key] != 1)
+                {
+                    *m_data[key] = 1;
+                    m_deleted++;
```

Comment:
suggest to rename to m_inserted. BTW, please consider to use 1 bit instead of 1 byte to represent one occurrence.